src/main/java/org/cloudfoundry/promregator/scanner/ReactiveTargetResolver.java

```diff
@@ -401,6 +422,40 @@ public void setResolvedApplicationId(String resolvedApplicationId) {
 			return itnew;
 		});
 	}
+
+	private Flux<IntermediateTarget> resolveAnnotations(IntermediateTarget it) {
+		if (Boolean.TRUE.equals(it.getConfigTarget().getKubernetesAnnotations())) {
+			Mono<org.cloudfoundry.client.v3.applications.ListApplicationsResponse> response = this.cfAccessor
+				.retrieveAllApplicationsInSpaceV3(it.getResolvedOrgId(), it.getResolvedSpaceId());
+
+			return response.flatMap(res -> {
+				if (res == null || INVALID_APPLICATIONS_RESPONSE == res) {
+					logEmptyTarget
+						.warn("Your foundation does not support V3 APIs, yet you have enabled Kubernetes Annotation filtering. Ignoring annotation filtering.");
+					return Mono.just(it);
+				}
+
+				return res.getResources().stream()
```

Comment:
I wasn't sure how to reformat this. I'm not super familiar with Flux. I could `Flux.fromIterable(res.getResources())` but at the end of this logic I want to get the first element or else return an empty Mono. I can `blockFirst()` with a timeout I guess? But that didn't seem very clean either.

Open to suggestions

Reply:
I had a closer look at this topic and came up with ff4b24c3b7a0046cc39a10f92aa2f6bb12274c45. 
To my mind, it does not bring up a lot of value to either stream here or to add too much of reactive programming sugar. What we are doing here is more or less just working on a set of objects that is anyhow in main memory. So, I would prefer to shout out for applying the [KISS principle](https://en.wikipedia.org/wiki/KISS_principle).

Reply:
In my opinion a stream is going to be simpler than doing a bunch of loops and if statements.

Reply:
Let me know if you'd rather have streams or I saw you did a commit to a branch I could bring in instead.

Reply:
For me `if` and `for` are simpler than streams, but I am not religious about that. We may keep it as it is. We don't lose/win much either way (I had thought it would have been more before).